cmd/thanos/store.go

```diff
@@ -107,6 +107,12 @@ func registerStore(app *extkingpin.App) {
 		"Default is 24h, half of the default value for --delete-delay on compactor.").
 		Default("24h"))
 
+	lazyIndexReaderEnabled := cmd.Flag("store.enable-index-header-lazy-reader", "If true, Store Gateway will lazy memory map index-header only once required by a query.").
```

Comment:
I think it should be default true at some point, so what about unhidding this flag?

```suggestion
	lazyIndexReaderEnabled := cmd.Flag("store.enable-index-header-lazy-reader", "If true, Store Gateway will lazy memory map index-header only once the block is required by a query.").
```

Reply:
Ok, unhiding it. I wasn't just that optimistic that this could be the default at some point (still need to test performances  on opening a very large block index).